mesonbuild/backend/ninjabackend.py

```diff
@@ -2920,10 +2920,11 @@ def generate_common_compile_args_per_src_type(self, target: build.BuildTarget) -
         return src_type_to_args
 
     def generate_single_compile(self, target: build.BuildTarget, src,
-                                is_generated=False, header_deps=None,
+                                is_generated: bool = False, header_deps=None,
                                 order_deps: T.Optional[T.List['mesonlib.FileOrString']] = None,
                                 extra_args: T.Optional[T.List[str]] = None,
-                                unity_sources: T.Optional[T.List[mesonlib.FileOrString]] = None) -> None:
+                                unity_sources: T.Optional[T.List[mesonlib.FileOrString]] = None
+                                ) -> T.Tuple[str, str]:
```

Comment:
If we're going to break this across lines (I don't love it but I don't hate it), then at least we should go ahead and include the trailing comma here...